src/resolvers/vertex.js

```diff
@@ -0,0 +1,126 @@
+const { combineResolvers } = require("graphql-resolvers");
+const {
+  isAuthenticated: isAuthenticatedResolver,
+  isVertexOwner: isVertexOwnerResolver,
+} = require("./authorization");
+const {
+  isAuthenticated,
+  isGraphOwner,
+  isVertexOwner,
+  Response,
+} = require("../utils/");
+
+const vertexResolvers = {
+  Query: {
+    vertex: combineResolvers(
+      isAuthenticatedResolver,
+      isVertexOwnerResolver,
+      async (root, { id }, { models, me }) => {
+        return models.vertex.findByPk(id);
+      }
+    ),
+  },
+
+  Mutation: {
+    async createVertex(root, { data, x, y, graphId }, { models, me }) {
+      const checkIsGraphOwner = await isGraphOwner(null, graphId, models, me);
+
+      if (!isAuthenticated(me)) {
+        const res = new Response("Log in to create new vertices", 403, false);
+        return { ...res, vertex: null };
+      } else if (!checkIsGraphOwner) {
+        const res = new Response(
+          "You're not the owner of this mind-map.",
+          403,
+          false
+        );
+        return { ...res, vertex: null };
+      } else {
+        const res = new Response("Vertex created.");
+        const newVertex = await models.vertex.create({
+          data,
+          x,
+          y,
+          graphId,
+        });
+        return { ...res, vertex: newVertex };
+      }
+    },
+
+    async updateVertexData(root, { id, data }, { models, me }) {
+      const checkIsVertexOwner = await isVertexOwner(id, null, models, me);
+
+      if (!isAuthenticated(me)) {
+        const res = new Response("Log in to update vertex.", 403, false);
+        return { ...res, vertex: null };
+      } else if (!checkIsVertexOwner) {
+        const res = new Response(
+          "You're not the owner of this vertex",
+          403,
+          false
+        );
+        return { ...res, vertex: null };
```

Comment:
Consider removing this bit of code to some outer function, this first if/elseif. You repeat it in several resolvers. you can make a function that is like 
```
const validateUserVertex = (id, models, me) => {
const checkIsVertexOwner = await isVertexOwner(id, null, models, me);
 // check on isAuthenticated(me) and `checkIsVertexOwner`
// return either a response (can be any of the two) or null
}
```

then from the resolver you do
```
const userInvalid = validateUserVertex(id, models, me)

if (userInvalid) {
 return { ...userInvalid, vertex: null }
} else {
const res = new Response("Vertex data updated.");
const updatedVertex = await models.vertex
          .findByPk(id)
          .then((vertex) => vertex.update({ data }));
return { ...res, vertex: updatedVertex };
}
```